### PR TITLE
Cleans up language_level_df

### PR DESCRIPTION
**Pull request by SimonGreenhill**
* refactors to match coding style
* removes unneeded method (drop_question_marks). People should do this themselves.
* ensures that we get a LanguageTable as input.
* fixes the tests

### CODE REVIEW

**Comment by HedvigS** [comment]
Reviewing now.

> * removes unneeded method (drop_question_marks). People should do this themselves.

I disagree, I think this is a good place to do this. It has a large effect on "singular_least_missing_data", and I'd rather there was a way of addressing this here rather than leaving users to not realise this and then backtrack.

**Comment by HedvigS** [comment]
Re [5310bf4](https://github.com/grambank/rgrambank/pull/22/commits/5310bf409a7518ef659f1ece0979cbb5eab2c673), does "refactor" means standardise whitespaces? Or is there something that went missing here?

**Comment by HedvigS** [comment]
I was going to add to this function by first matching Language_ID in ValueTable to ID in LanguageTable and then do the stuff with Glottocode and Language_level_ID. That way it can be used on cldf-datasets that don't happen to be blessed with having Glottocodes as Language_IDs ^^!

I'll do that once this is merged.

File: R/language_level_df.R [hunk 1]
```diff
@@ -1,134 +1,101 @@
...
-}
-
-# if there is a missing language level ID, which it can be in some datasets where only dialects get language level IDs and languages and families don't, then replace those with the content in the Language_ID column.
-LanguageTable   <- LanguageTable %>%
-  dplyr::mutate(Language_level_ID = ifelse(is.na(Language_level_ID)|
-                                               Language_level_ID == "", Language_ID, Language_level_ID))
-
-## QUESTION MARK ACTION
-
-#turn question marks into missing values if set to TRUE
```
Comment (HedvigS) [inline R/language_level_df.R:55]: I think we should keep the part of this function that lets the user turn ? into NAs or not.

**Review by HedvigS — CHANGES_REQUESTED** [review]

File: R/language_level_df.R (hunk 1 above)
Comment (SimonGreenhill) [reply R/language_level_df.R:55]: Refactor means "tidy up and improve".

I removed the question mark thing as it is out of scope here and feels wrong for a few reasons: 
1. This is a method to make a language level data frame. Magically transforming data is  something else. 
2. It's better for people to operate on the original data and decide what to do with "?" explicitly. 
3. It's badly named (what's a question mark? "drop" means remove, but this recodes).
4. ideally rcldf should handle this at the CLDF level (https://github.com/SimonGreenhill/rcldf/issues/14)
5. It's excessive. If we add these sorts of features to the R package it's more code we need to maintain and we need to test. Users can do this easily using tools already available to them, so why do we need to put effort into implementation and maintenance:

```{r}
values <- gb$tables$ValueTable %>% mutate(Value = na_if(Value, "?"))

# or even:
lld <- gb$tables$ValueTable %>%
    mutate(Value = na_if(Value, "?")) %>%
    language_level_df(., gb$tables$LanguageTable, 'singular_least_missing_data')
```

Comment (HedvigS) [reply R/language_level_df.R:55]: Okay, I am not fully convinced but it's not a hill I'm gonna die on. Let's do it your way.

**Review by HedvigS — APPROVED** [review]

**Pull request merged**